R/update-news.R

```diff
@@ -354,6 +355,35 @@ default_type <- function() {
   "Uncategorized"
 }
 
+capitalize_description <- function(df) {
+  df$description
+
+  # leave code alone
+  if (substr(df$description, 1, 1) == "`") {
+    return(df)
+  }
+
+  # leave package name alone
+  pkg_name <- desc::desc_get("Package")
+  start_with_pkg <- (grepl(sprintf("^%s ", pkg_name), df$description) ||
+    grepl(sprintf("^%s's", pkg_name), df$description))
+  if (start_with_pkg) {
+    return(df)
+  }
+
+  df$description <- paste0(
+    toupper(substr(df$description, 1, 1)),
+    substr(df$description, 2, nchar(df$description))
+  )
```

Comment:
Surprisingly tricky. I was thinking about a rematch2 solution, which could also include the `grepl()` logic from above, and allow for vectorization (and more compact code) perhaps?

"Bury the logic in one complex regex..." 🤔

Reply:
I'm re-reading `?toupper` and actually one of the examples there is very similar to what I did above (which does not mean there are no better solutions!).

Reply:
One simple solution would be to add a dependency on https://cran.r-project.org/web/packages/snakecase/index.html but that might be overkill for one call only.